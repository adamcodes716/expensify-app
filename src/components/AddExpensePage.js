import React from 'react';

const AddExpensePage= () => {
    return <div>
       This is from my Add ExpenseComponent!
    </div>
}

export default AddExpensePage;