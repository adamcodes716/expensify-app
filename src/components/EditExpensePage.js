import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage= (props) => {
    return ( 
    <div>
       <ExpenseForm
         expense={props.expense}
          onSubmit={(expense) => {
              //Dispatch the action to edit the expense
              props.dispatch(editExpense(props.expense.id, expense));
              //Redirect to the dashboard
              props.history.push('/'); // will use routing and then not full refresh
            console.log('updated', expense);
        } } 
       />
       <button onClick= {() => {
         props.dispatch(removeExpense({ id: props.expense.id })); // shorthane
         props.history.push('/'); // will use routing and then not full refresh
        }}
        >Remove</button>
    </div>
    );  
};

const mapStateToProps = (state, props) => {
   return {
       expense: state.expenses.find((expense) => { // could also convert this to shorthand
           return expense.id === props.match.params.id;
       })
   }
};

export default connect(mapStateToProps)(EditExpensePage);
//export default EditExpensePage;