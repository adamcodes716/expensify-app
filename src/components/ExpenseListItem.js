import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
    <div>
        <h3>{description}</h3>
        <p>{amount} - {createdAt}</p>
        <button onClick= {() => {
            dispatch(removeExpense({ id: id })) // could use shorthand here
        }}
        >Remove</button>
    </div>
);

export default connect()(ExpenseListItem);