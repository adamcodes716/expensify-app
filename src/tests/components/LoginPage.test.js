import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const startLogin = jest.fn();  // create a spy
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);  // pass in the spy
  wrapper.find('button').simulate('click');  // click the button
  expect(startLogin).toHaveBeenCalled();  
});
