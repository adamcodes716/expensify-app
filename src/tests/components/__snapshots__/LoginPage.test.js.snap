// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div>
  <h1>
    Login Page
  </h1>
  <input
    autoFocus={true}
    placeholder="User Name"
    type="text"
  />
  <input
    placeholder="Password"
    type="text"
  />
</div>
`;
